kore/src/Kore/Variables/Fresh.hs

```diff
@@ -223,3 +225,37 @@ refreshVariables avoid0 =
         -- The variable does not collide with any others, so renaming is not
         -- necessary.
         (Set.insert var avoid, rename)
+
+generateFreshVars
+    :: forall variable
+     . FreshVariable variable
+    => FreshPartialOrd variable
+    => SortedVariable variable
+    => From Variable variable
+    => Set (ElementVariable variable)
+    -- ^ variables to avoid
+    -> Text.Text
+    -- ^ base name for variables to be generated
+    -> [Sort]
+    -- ^ sorts for the variables to be generated
+    -> [ElementVariable variable]
+generateFreshVars avoid0 base sorts =
+    snd $ foldr worker (avoid0, []) indexedSorts
+  where
+    indexedSorts = zip [1..] sorts
```

Comment:
If we use these counters and sorts to initialize some `ElementVariable`s, then we can re-use `refreshVariables` here.

Reply:
We could do that, but it seemed more convoluted. because then I'd need to get the list of variables to be able to quantify them... But I'll try an do that.

Reply:
I've changed this into an freshSymbolInstance generating a TermLike in TermLike.hs.